### PR TITLE
allow modifying tags when query string is already provided on URL

### PR DESCRIPTION
**Pull request by headphonejames**
## Hey, I just made a Pull Request!

Fixed bug https://github.com/backstage/backstage/issues/17021

Tags could not be modified when the catalog page was refreshed with a query string with previously selected items. 

https://user-images.githubusercontent.com/3026841/227609076-e7ac29f7-3f3f-4b12-a6f8-caa0235a9e19.mov

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Parsifal-M — COMMENTED** [review]
Nice one! :rocket:

**Comment by catmeme** [comment]
Can this get some attention?  We would love to have this bug addressed.  Thank you!

File: plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.tsx [hunk 1]
```diff
@@ -96,14 +95,12 @@ export function EntityAutocompletePicker<
...
-    ) {
-      setSelectedOptions(queryParameters);
-    }
-  }, [selectedOptions, queryParameters]);
+  const [prevQueryParameters, setQueryParameter] = useState(queryParameters);
+  // if the query parameter has changed, update the selected options
+  if (queryParameters !== prevQueryParameters) {
+    setSelectedOptions(queryParameters);
+    setQueryParameter(queryParameters);
+  }
```
Comment (Rugvip) [inline plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.tsx:103]: Mutating state directly on render should be avoided, we'll want to keep this wrapped in `useEffect` or some other solution. Does simply removing the `selectedOptions` from the existing `useEffect` work?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Sorry this took time to review

File: plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.tsx (hunk 1 above)
Comment (catmeme) [reply plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.tsx:103]: @headphonejames are you able to update this PR to align with the requested changes?

Comment (headphonejames) [reply plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.tsx]: I updated the change and verified that it is working.

File: .changeset/smart-crabs-dream.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-react': minor
+---
+
+Reverted the check if the selected options list is different than the query parameters list before invoking setSelectedOptions method. This was preventing updating list items when a query string was already present in the URL when loading the page.
```
Comment (Rugvip) [inline .changeset/smart-crabs-dream.md:5]:
```suggestion
---
'@backstage/plugin-catalog-react': patch
---

Reverted the check if the selected options list is different than the query parameters list before invoking `setSelectedOptions` method. This was preventing updating list items when a query string was already present in the URL when loading the page.
```

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 

Just shipping a small changeset tweak

**Pull request merged**